respect-app-compose/src/commonMain/kotlin/world/respect/app/view/clazz/detail/ClazzDetailScreen.kt

```diff
@@ -0,0 +1,265 @@
+import world.respect.shared.viewmodel.clazz.detail.ClazzDetailViewModel
+
+@Composable
+fun ClazzDetailScreen(
+    viewModel: ClazzDetailViewModel
+) {
+    val uiState by viewModel.uiState.collectAsState()
+
+    ClazzDetailScreen(
+        uiState = uiState,
+        onClickAddPersonToClazz = viewModel::onClickAddPersonToClazz,
+        onClickSortOption = { viewModel.onClickSortOption(it) },
+        onSelectChip = { viewModel.onSelectChip(it) },
+    )
+}
+
+@Composable
+fun ClazzDetailScreen(
+    uiState: ClazzDetailUiState,
+    onClickAddPersonToClazz: () -> Unit,
+    onClickSortOption: (String) -> Unit,
+    onSelectChip: (String) -> Unit
+) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        item {
+
+            RespectFilterChipsHeader(
+                options = uiState.chipOptions.map{it.option},
+                selectedOption = uiState.selectedChip,
+                onOptionSelected = { onSelectChip(it) },
+                optionLabel = { it }
+            )
+
+            RespectSortHeader(
+                options = uiState.sortOptions.map { it.option },
+                selectedOption = uiState.selectedSortOption,
+                onOptionSelected = onClickSortOption,
+                optionLabel = { it },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 12.dp)
+            )
+        }
+
+        item {
+            ListItem(
+                headlineContent = {
+                    Text(
+                        text = stringResource(
+                            resource = Res.string.pending_invite
+                        )
+                    )
+                },
+                trailingContent = {
+                    Icon(
+                        imageVector = Icons.Outlined.KeyboardArrowDown,
+                        contentDescription = null,
+                        modifier = Modifier.size(24.dp),
+                    )
+                }
+            )
+        }
+
+        itemsIndexed(
+            uiState.listOfPending,
+            key = { index, pendingUser ->
+                pendingUser.sourcedId
+            }) { index, pendingUser ->
+            val roleName = pendingUser.roles.firstOrNull()?.role?.name
+                ?.lowercase()
+                ?.replaceFirstChar { it.uppercase() } ?: "Unknown"
+
+            ListItem(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                leadingContent = {
+                    RespectPersonAvatar(
+                        name = pendingUser.givenName,
+                        modifier = Modifier.size(40.dp),
+                        fontScale = 1.0f
+                    )
+                },
+                headlineContent = {
+                    Text(text = pendingUser.givenName + "(" + roleName + ")")
+                },
+                trailingContent = {
+                    Row {
+                        Icon(
```

Comment:
This obviously needs an event handler - needs to go back to the viewmodel

Reply:
Event handler was added; but it was missing a parameter (e.g. the user to be accepted/rejected), which was also (obviously) needed.